### PR TITLE
Photon raytracing domain extension v0.0.2

### PR DESCRIPTION
**Pull request by aljosahafner**
*Please add a brief description (one sentence) here and link the issue this pull-request implements*

*Implements issue:* #...

## Description

*As detailed description as possible.*

*What is introduced, removed or renamed and why?*
*What is made required, recommended, optional?*
*What concept stands behind this change?*

*Please also add an example.*

## Affected Components

- `base`
- `EXT-...`

## Logic Changes

*Which logic changes are conceptually introduced?*

## Writer Changes

*How does this change affect data writers?*
*What would a writer need to change?*
*Does this pull request change the interpretation of existing data writers?*

- `libopenPMD` (upcoming): https://github.com/ComputationalRadiationPhysics/libopenPMD/...

## Reader Changes

*How does this change affect data readers?*

*What would a reader need to change? Link implementation examples!*

- `openPMD-validator`: https://github.com/openPMD/openPMD-validator/...
- `openPMD-viewer`: https://github.com/openPMD/openPMD-viewer/...
- `yt`: https://github.com/yt-project/yt/...
- `VisIt`: https://github.com/openPMD/openPMD-visit-plugin
- `libopenPMD` (upcoming): https://github.com/ComputationalRadiationPhysics/libopenPMD/...
- `XDMF` (upcoming): https://github.com/openPMD/openPMD-tools/...

## Data Converter

*How does this affect already existing files with previous versions of the standard?*
*Is this change possible to be expressed in a way, that an automated tool can update the file?*
*Link code/pull requests that implement the upgrade logic!*

- https://github.com/openPMD/openPMD-converter/...

### CODE REVIEW

File: EXT_PRAYTRACE.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+- `date`: `10.10.2019 00:00:00 +0100`
+- `nRays`
+	- type: *(uintX)*
+	- description: Number of initialised rays
+	- scope: *required*
+	- unitSI: 1.0
+	- unitDimension: [unitless] (0., 0., 0., 0., 0., 0., 0.)
+
+### Structure
+data/1/particlesPath/rays/0...N/position...
```
Comment (CFGrote) [inline EXT_PRAYTRACE.md]: if ray numbering is 0 based, last should be N-1, right?

File: EXT_PRAYTRACE.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+		- `direction/z`
+			- type: *(floatX)*
+			- description: *x-* direction
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [m] (1., 0., 0., 0., 0., 0., 0.)
+
+	- `eFieldSPolarisation/`
+		- `eFieldSPolarisation/x`
+			- type: *(complexX)*
```
Comment (CFGrote) [inline EXT_PRAYTRACE.md]: why complexX? is float not good enough?
a more descriptive name may be "S polarized E field, x component"

File: EXT_PRAYTRACE.md (hunk 1 above)
Comment (CFGrote) [inline EXT_PRAYTRACE.md]: if ray numbering is 0 based, last should be N-1, right?

File: EXT_PRAYTRACE.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+			- unitSI: 1.0
+			- unitDimension: [unitless] (0., 0., 0., 0., 0., 0., 0.)
+		- `phase/pPol`
+			- type: *(complexX)*
+			- description: phase for P polarisation
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [unitless] (0., 0., 0., 0., 0., 0., 0.)
+	
+### Additional Attributes for the Group `particlesPath`
```
Comment (CFGrote) [inline EXT_PRAYTRACE.md]: i would list all attributes together and indicate in the scope whether required or optional

**Review by CFGrote — CHANGES_REQUESTED** [review]
Hi Aljosa,
there's now one redundant level in the hierarchy, e.g
data/1/particles/rays/1/position/position/x, just data/1/particles/rays/1/position/x will do.

File: EXT_PRAYTRACE.md (hunk 2 above)
Comment (aljosahafner) [reply EXT_PRAYTRACE.md]: You're right...

File: EXT_PRAYTRACE.md (hunk 1 above)
Comment (aljosahafner) [reply EXT_PRAYTRACE.md]: Hey, I agree, I'll correct it. Regarding the hierarchy, I followed the example from NRAYTRACING, where the attributes are also marked like that: 'position/' - 'position/x'

**Pull request closed without merging**